Documentation/coding-guidelines/project-guidelines.md

```diff
@@ -1,50 +1,84 @@
+ - `$(PackageTargetFramework) -> netstandard1.x or netcoreapp1.x or net46x or uap10.x`
+ - `$(PackageTargetRuntime) -> empty or aot`
+ - For more information on various targets see also [.NET Standard](https://github.com/dotnet/standard/blob/master/docs/versions.md)
 
-In the pkg directory for the library there should be only **one** `.pkgproj` for the primary package for the library.  If the library has platform-specific implementations those should be split into platform specific projects in a subfolder for each platform. (see [Package projects](./package-projects.md))
+<BR/>`*` -> *default values*
 
-##Build Pivots
-Below is a list of all the various options we pivot the project builds on.
+##Aggregate build properties
+Each project will define a set of supported build configurations
 
-- **Architecture:** x86, x64, ARM, ARM64
-- **Flavor:** Debug, Release
-- **OS:** Windows_NT, Linux, OSX, FreeBSD
-- **Platform Runtimes:** NetFx (aka CLR/Desktop), CoreCLR, CoreRT (aka NetNative/AOT/MRT)
-- **Target Frameworks:** NetFx (aka Desktop), netstandard (aka dotnet/Portable), NETCoreApp (aka .NET Core), UAP (aka UWP/Store/netcore50)
-- **Version:** Potentially multiple versions at the same time.
-- **TestTFM:** net46, netcoreapp1.1, netcore50. This is the TFM that will be used to run the tests in. (Used in test projects only)
+```
+<PropertyGroup>
+  <BuildConfigurations>
+    [BuildConfiguration]
+  </BuildConfigurations>
+<PropertyGroup>
+```
 
-##Full Repo build pass
-**Build Parameters:** *Flavor, Architecture*<BR/>
-For each combination of build parameters there should be a full build pass over the entire repo.
+- `BuildConfiguration -> $(Configuration)|$(Platform)`
+- `$(Configuration) -> $(OSGroup)-$(TargetGroup)-$(ConfigurationGroup)`
 
-##Project build pass
-**Optional Build Parameters:** *OS, Platform Runtime, Target Framework, Version*<BR/>
-These are optional build parameters for specific projects and don't require a full build pass but instead should be scoped to individual projects within the most appropriate full build pass.
+##Full build configuration
+The `BuildConfiguration` property can either be set directly via
+`/p:BuildConfiguration=[BuildConfiguration]` or will be defaulted based
+on where you are running the build (i.e. `netcoreapp|Windows_NT`).
 
-#Project configuration conventions
+##Supported full build configurations
+- .NET Core latest on current OS (default) -> netcoreapp|[RunningOS]
+- .NET Framework -> netfx|Windows_NT
+- UWP -> uap|Windows_NT
+Open question: What about other build configurations like netcoreapp|aot or uap|F5?
+
+##Project configuration conventions
 For each unique configuration needed for a given library project a configuration property group should be added to the project so it can be selected and built in VS and also clearly identify the various configurations.<BR/>
 `<PropertyGroup Condition="'$(Configuration)|$(Platform)' == '$(<OSGroup>_<TargetGroup>_<ConfigurationGroup>)|$(Platform)'">`
```

Comment:
We can definitely do that but until we have a tool we should follow the convention.